### PR TITLE
VCS tracer: Support relative paths, fix contains field

### PR DESCRIPTION
**Pull request by kyleam**
This PR

  * adds a test fixture for git repos

    It will probably turn out to be too simple, but it seems ok for
    now.  As a result, travis should now be testing the change from
    #165.

  * moves the VCS tests into a distributions/tests file (like the
    Debian/conda/virtualenv tracers)

  * adjust the VCS tracers to support relative paths

    Simply expanding to the full path seems to work, but perhaps I'm
    missing edge cases/complications.

   * fixes the contains=True issue from #166

### CODE REVIEW

**Comment by yarikoptic** [comment]
try
```shell
$> ln -s /tmp /tmp/symlinkedtmp
$> TMPDIR=/tmp/symlinkedtmp python -m pytest -s -v niceman/distributions/tests/test_vcs.py  
```
and welcome to my world of self-imposed misery and suffering as a sacrifice to the humanity so others do not have to go through those pains ;)

File: niceman/distributions/tests/test_vcs.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+from niceman.tests.utils import assert_is_subset_recur
+from niceman.tests.fixtures import git_repo_fixture
+
+
+git_repo_empty = git_repo_fixture(kind="empty")
+git_repo = git_repo_fixture()
+git_repo_pair = git_repo_fixture(kind="pair")
+
+
+# TODO: Move to niceman.test.utils and use in other tracer tests.
```
Comment (yarikoptic) [inline niceman/distributions/tests/test_vcs.py]: +1

File: niceman/distributions/tests/test_vcs.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        os.path.join(git_repo, "foo"),
+        # ... and relative paths work.
+        "bar",
+        # So do paths in subdirectories.
+        os.path.join(git_repo, "subdir/baz")
+    ]
+
+    tracer = VCSTracer()
+
+    with chpwd(git_repo):
```
[the remark this answers is not in the document]
Comment (kyleam) [reply niceman/distributions/tests/test_vcs.py]: I think it is necessary because the "bar" path above is relative.
